### ISSUE TITLE
[Bug]: the result of  tpch Q16 is not correct

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#3206]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.50
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

the correct result should be :
mysql> select
    ->     p_brand,
    ->     p_type,
    ->     p_size,
    ->     count(distinct ps_suppkey) as supplier_cnt
    -> from
    ->     partsupp,
    ->     part
    -> where
    ->     p_partkey = ps_partkey
    ->     and p_brand <> 'Brand#35'
    ->     and p_type not like 'ECONOMY BURNISHED%'
    ->     and p_size in (14, 7, 21, 24, 35, 33, 2, 20)
    ->     and ps_suppkey not in (
    ->         select
    ->             s_suppkey
    ->         from
    ->             supplier
    ->         where
    ->             s_comment like '%Customer%Complaints%'
    ->     )
    -> group by
    ->     p_brand,
    ->     p_type,
    ->     p_size
    -> order by
    ->     supplier_cnt desc,
    ->     p_brand,
    ->     p_type,
    ->     p_size
    -> ;
+----------+---------------------------+--------+--------------+
| p_brand  | p_type                    | p_size | supplier_cnt |
+----------+---------------------------+--------+--------------+
| Brand#11 | STANDARD PLATED TIN       |     20 |            4 |
| Brand#12 | MEDIUM PLATED STEEL       |     20 |            4 |
| Brand#13 | PROMO BURNISHED COPPER    |      7 |            4 |
| Brand#13 | SMALL BRUSHED COPPER      |     20 |            4 |
| Brand#14 | ECONOMY POLISHED STEEL    |     24 |            4 |
| Brand#14 | LARGE ANODIZED TIN        |     20 |            4 |
| Brand#15 | LARGE POLISHED TIN        |     20 |            4 |
| Brand#15 | STANDARD PLATED STEEL     |     24 |            4 |
| Brand#21 | SMALL BRUSHED COPPER      |     35 |            4 |
| Brand#22 | STANDARD ANODIZED TIN     |     35 |            4 |
| Brand#23 | ECONOMY ANODIZED TIN      |      7 |            4 |
| Brand#23 | LARGE PLATED BRASS        |     35 |            4 |
| Brand#23 | SMALL ANODIZED NICKEL     |     33 |            4 |
| Brand#24 | LARGE ANODIZED TIN        |      2 |            4 |
| Brand#31 | STANDARD PLATED STEEL     |     20 |            4 |
| Brand#32 | PROMO PLATED TIN          |      2 |            4 |
| Brand#33 | MEDIUM ANODIZED COPPER    |     35 |            4 |
| Brand#33 | PROMO PLATED COPPER       |      7 |            4 |
| Brand#33 | STANDARD BURNISHED NICKEL |      7 |            4 |
| Brand#34 | SMALL PLATED BRASS        |     14 |            4 |
| Brand#42 | STANDARD POLISHED BRASS   |     21 |            4 |
| Brand#43 | ECONOMY BRUSHED STEEL     |     20 |            4 |
| Brand#43 | SMALL POLISHED NICKEL     |      2 |            4 |
| Brand#45 | LARGE BURNISHED BRASS     |     14 |            4 |
| Brand#45 | STANDARD POLISHED STEEL   |     24 |            4 |
| Brand#51 | ECONOMY POLISHED STEEL    |      7 |            4 |
| Brand#52 | MEDIUM PLATED STEEL       |     20 |            4 |
| Brand#52 | MEDIUM POLISHED BRASS     |      2 |            4 |
| Brand#52 | SMALL BURNISHED NICKEL    |     14 |            4 |
| Brand#53 | MEDIUM POLISHED TIN       |      2 |            4 |
| Brand#53 | PROMO BRUSHED COPPER      |     24 |            4 |
| Brand#53 | PROMO POLISHED NICKEL     |     33 |            4 |
| Brand#21 | MEDIUM BURNISHED STEEL    |     24 |            2 |
| Brand#22 | SMALL PLATED STEEL        |      2 |            2 |
| Brand#32 | MEDIUM BRUSHED STEEL      |      7 |            2 |
| Brand#42 | SMALL POLISHED STEEL      |     35 |            2 |
| Brand#43 | MEDIUM ANODIZED BRASS     |     14 |            2 |
| Brand#45 | STANDARD BURNISHED BRASS  |      2 |            2 |
| Brand#52 | PROMO POLISHED STEEL      |     35 |            1 |
| Brand#53 | ECONOMY BRUSHED TIN       |     21 |            1 |
| Brand#53 | STANDARD BURNISHED STEEL  |      7 |            1 |
+----------+---------------------------+--------+--------------+
41 rows in set (0.00 sec)

but actual is:
mysql> select
    ->     p_brand,
    ->     p_type,
    ->     p_size,
    ->     count(distinct ps_suppkey) as supplier_cnt
    -> from
    ->     partsupp,
    ->     part
    -> where
    ->     p_partkey = ps_partkey
    ->     and p_brand <> 'Brand#35'
    ->     and p_type not like 'ECONOMY BURNISHED%'
    ->     and p_size in (14, 7, 21, 24, 35, 33, 2, 20)
    ->     and ps_suppkey not in (
    ->         select
    ->             s_suppkey
    ->         from
    ->             supplier
    ->         where
    ->             s_comment like '%Customer%Complaints%'
    ->     )
    -> group by
    ->     p_brand,
    ->     p_type,
    ->     p_size
    -> order by
    ->     supplier_cnt desc,
    ->     p_brand,
    ->     p_type,
    ->     p_size
    -> ;
Empty set (0.01 sec)

**Comment by nnsgmsone** [comment]
fixed by #3206

**Comment by aressu1985** [comment]
fixed
